packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -357,37 +357,46 @@ function WorklistDataGrid(props) {
     }
   };
 
-  const selectMultipleRows = useCallback((newEndRowIndex, rangeInfo) => {
-    if (!onRowSelect || !rangeInfo) {
-      return;
+  const getRowsInRangeToUnselect = useCallback((startOfRange, prevEndOfRange, newEndOfRange) => {
+    let rowIdsToUnselect = [];
+    if (!prevEndOfRange) {
+      // short-circuit since there was no previous range, there is nothing to unselect.
+      return rowIdsToUnselect;
+    }
+
+    if (startOfRange > prevEndOfRange) {
+      // The range extends upward from the anchor row
+      if (newEndOfRange > prevEndOfRange) {
+        // The range was moved down towards the anchor so rows that no longer qualify for the range need to be unselected.
+        rowIdsToUnselect = rows.slice(prevEndOfRange - 1, Math.min(newEndOfRange, startOfRange)).map(row => row.id);
+      }
+    } else if (startOfRange < prevEndOfRange) {
```

Comment:
I would think that this could be an else condition.  We already checked for greater than in the if condition.

Reply:
What about the case when they are equal?

Reply:
Saad is correct, if we use an else, the equal case would get merged into the else block which technically would not be correct. The explicit checks are also there to show the direction of the range from the anchor.